index.js

```diff
@@ -63,621 +63,623 @@ var equal = function equal(query, field, value) {
-            var _result = slicedToArray(result, 1);
-
-            var records = _result[0];
-
-            records = records.filter(function (record) {
-                return record;
-            });
-            if (records.length) {
-                props = jsData.utils.plainCopy(props);
-                return _this11._updateHelper(mapper, records, records.map(function () {
-                    return props;
-                }), opts);
-            }
-            return [[], {}];
-        });
-    },
-
-
-    /**
-     * Update the given records in a single batch. Internal method used by
-     * Adapter#updateMany.
-     *
-     * @method DataStoreAdapter#_updateMany
-     * @private
-     * @param {Object} mapper The mapper.
-     * @param {Object[]} records The records to update.
-     * @param {Object} [opts] Configuration options.
-     * @return {Promise}
-     */
-    _updateMany: function _updateMany(mapper, records, opts) {
-        var _this12 = this;
-
-        records || (records = []);
-        var idAttribute = mapper.idAttribute;
-        var tasks = records.map(function (record) {
-            return _this12._find(mapper, jsData.utils.get(record, idAttribute), opts);
-        });
-        return jsData.utils.Promise.all(tasks).then(function (results) {
-            var _records = results.map(function (result) {
-                return result[0];
-            });
-            _records.forEach(function (record, i) {
-                if (!record) {
-                    records[i] = undefined;
-                }
-            });
-            _records = _records.filter(function (record) {
-                return record;
-            });
-            records = records.filter(function (record) {
-                return record;
-            });
-            if (_records.length) {
-                records = jsData.utils.plainCopy(records);
-                return _this12._updateHelper(mapper, _records, records, opts);
-            }
-            return [[], {}];
+        return err ? reject(err) : resolve([entity ? entity : undefined, {}]);
+      });
+    });
+  },
+
+
+  /**
+   * Retrieve the records that match the selection query. Internal method used
+   * by Adapter#findAll.
+   *
+   * @method DataStoreAdapter#_findAll
+   * @private
+   * @param {Object} mapper The mapper.
+   * @param {Object} [query] Selection query.
+   * @param {Object} [opts] Configuration options.
+   * @return {Promise}
+   */
+  _findAll: function _findAll(mapper, query, opts) {
+
+    var _this7 = this;
+    var meta = {};
+
+    return new jsData.utils.Promise(function (resolve, reject) {
+      try {
+        var dsQuery = _this7.datastore.createQuery(_this7.getKind(mapper, opts));
+        dsQuery = _this7.filterQuery(dsQuery, query, opts);
+        _this7.datastore.runQuery(dsQuery, function (err, entities) {
+          if (err) {
+            return reject(err);
+          }
+          entities = [entities];
```

Comment:
No check for it being an array already?

Reply:
This fix is not to change findAll function, but to fix an arror: if entities is already an array, in jsdatastore-adapter it will be sliced in 2 element because it expect to have an array like this:
result[
  entities,
  metadata
].
entities = [entities]; just put the results inside the first cell of results. Next step will be to retrieve metadata from Gcloud and put in the second cell.